File: backend/src/main/java/com/example/ddmdemo/dto/SearchDTO.java
package com.example.ddmdemo.dto;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class SearchDTO {
    private String employeeName;
    private String employerSurname;
    private String governmentName;
    private String governmentLevel;
    private String fullText;
    private String booleanQuery;
    private boolean lawDoc;
    private boolean contractDoc;
    private String typeOfSearch;
    private double radius;
    private String address;
}

File: frontend/udd-front/src/models/SearchDTO.ts
export class SearchDTO {
  constructor(
    public employeeName: string = '',
    public employerSurname: string = '',
    public governmentName: string = '',
    public governmentLevel: string = '',
    public fullText: string = '',
    public booleanQuery: string = '',
    public lawDoc: boolean = true,
    public contractDoc: boolean = true,
    public typeOfSearch: string = 'standard_search',
    //TODO: geo location 
  ) {}
}